libs/libarchfpga/src/physical_types_util.cpp

```diff
@@ -66,79 +63,96 @@ static int get_logical_block_physical_pin_num_offset(t_physical_tile_type_ptr ph
 
 static int get_pin_logical_num_from_pin_physical_num(t_physical_tile_type_ptr physical_tile, int physical_num);
 
-static int get_class_physical_num_from_class_logical_num(t_physical_tile_type_ptr physical_tile,
-                                                         const t_sub_tile* curr_sub_tile,
-                                                         t_logical_block_type_ptr curr_logical_block,
-                                                         int curr_relative_cap,
-                                                         int logical_class_num);
+static std::vector<int> get_pb_pin_src_pins(t_physical_tile_type_ptr physical_type,
+                                            const t_sub_tile* sub_tile,
+                                            t_logical_block_type_ptr logical_block,
+                                            int relative_cap,
+                                            const t_pb_graph_pin* pin);
 
-static std::vector<const t_pb_graph_pin*> get_pb_graph_node_pins(const t_pb_graph_node* pb_graph_node);
+static std::vector<int> get_pb_pin_sink_pins(t_physical_tile_type_ptr physical_type,
```

Comment:
Add a comment to explain: the functionality of the function , the inputs and the outputs.